### PR TITLE
Data Feed list preview

### PR DESCRIPTION
**Pull request by shivangi-singh1**
### :nut_and_bolt: Description: What code changed, and why?

- Data feed table is created
- List preview is enabled
- Sorting  is enabled
- Delete and Enable/Disable functionality is enabled.

### :chains: Related Resources 
Data Feeds UI: S3: List Preview of Integrations #5417 

### :+1: Definition of Done

- Data feed table should be visible
- Sorting should be performed on all columns namely Data feed name , Service , Integration type.
- checkbox to show/hide Service , Integration type should be functional.
- Delete data feed should be functional.
- Enable/disable on the data feed should be functional.
- If no data is available , No data feed preview page should be visible.

### :athletic_shoe: How to Build and Test the Change 
STEP 1
inside the hab studio

[default:/src:0]# build components/automate-ui-devproxy/
[default:/src:0]# start_automate_ui_background
[default:/src:0]# start_all_services

STEP 2
open new window
go to automate UI path

$ cd components/automate-ui
and run the command 

make serve

To run cypress test
cd e2e
npm run cypress:local

Type feat to enable data-feed
Navigate to settings/data-feeds

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable

image

Screenshot 2021-09-07 at 1 54 11 AM

### CODE REVIEW

File: components/automate-ui/src/app/app.module.ts [hunk 1]
```diff
@@ -209,6 +209,8 @@ import { WelcomeModalComponent } from './page-components/welcome-modal/welcome-m
 import { WarningBannerComponent } from './page-components/warning-banner/warning-banner.component';
 import { AppConfigService } from 'app/services/app-config/app-config.service';
 import { DataFeedCreateComponent } from './pages/data-feed-create/data-feed-create.component';
+import { DataFeedTableComponent
+} from './page-components/data-feed-table/data-feed-table.component';
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/app.module.ts:213]:
```javascript
import { 
   DataFeedTableComponent
} from './page-components/data-feed-table/data-feed-table.component';
```

You can write it like this as this is like the universal syntax we are following

File: components/automate-ui/src/app/entities/destinations/destination.model.ts [hunk 2]
```diff
@@ -4,7 +4,17 @@ export interface Destination {
   url: string;
   secret: string;
   enable?: boolean;
-  integration_type?: string;
-  meta_data?: string;
+  integration_types?: string;
+  meta_data?: [];
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/entities/destinations/destination.model.ts:8]:
```suggestion
  meta_data?: Array<KVData>;
```

File: components/automate-ui/src/app/entities/destinations/destination.model.ts [hunk 3]
```diff
@@ -4,7 +4,17 @@ export interface Destination {
...
   services?: string;
 }
+export interface Metadata {
+  key?: string;
+  value?: string;
+}
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/entities/destinations/destination.model.ts:14]: we already have KVData interface

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  constructor(
+    private store: Store<NgrxStateAtom>
+  ) {}
+
+  ngOnInit() {
+  }
+
+  ngOnChanges() {
+
+  }
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts:49]: please remove these

**Comment by vivekshankar1** [comment]
Please update the PR description accordingly

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts (hunk 4 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts:49]: Done

File: components/automate-ui/src/app/entities/destinations/destination.model.ts (hunk 2 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/entities/destinations/destination.model.ts:8]: Done

File: components/automate-ui/src/app/entities/destinations/destination.model.ts (hunk 3 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/entities/destinations/destination.model.ts:14]: Done

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+import { filter, takeUntil } from 'rxjs/operators';
+import { EntityStatus, pending } from 'app/entities/entities';
+import { destinationEnableStatus } from 'app/entities/destinations/destination.selectors';
+
+@Component({
+  selector: 'app-data-feed-table',
+  templateUrl: './data-feed-table.component.html',
+  styleUrls: ['./data-feed-table.component.scss']
+})
+export class DataFeedTableComponent implements OnInit, OnChanges {
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts:21]:
```suggestion
export class DataFeedTableComponent {
```
We should remove OnInit and OnChanges if we are not implementing them

Comment (shivangi-singh1) [reply components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.ts:21]: Done

**Comment by shivangi-singh1** [comment]
> Please update the PR description accordingly

Done

**Comment by vivek-yadav** [comment]
@shivangi-singh1 Please add screen shots of your output screens, and sign-off your commits. And Ensure all the pipeline is green.

File: components/automate-ui/src/app/entities/destinations/destination.model.ts [hunk 6]
```diff
@@ -1,10 +1,19 @@
...
   secret: string;
   enable?: boolean;
-  integration_type?: string;
-  meta_data?: string;
+  integration_types?: string;
+  meta_data?: Array<KVData>;
   services?: string;
 }
+
+
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/entities/destinations/destination.model.ts:14]: remove extra lines

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+    </div>
+</div>
+  <app-delete-object-modal
+    [visible]="deleteModalVisible"
+    objectNoun="data feed"
+    [objectName]="dataFeedToDelete?.name"
+    (close)="closeDeleteModal()"
+    (deleteClicked)="deleteDataFeed()"
+    objectAction="Delete">
+  </app-delete-object-modal>
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html:79]:
```suggestion
<app-delete-object-modal
  [visible]="deleteModalVisible"
  objectNoun="data feed"
  [objectName]="dataFeedToDelete?.name"
  (close)="closeDeleteModal()"
  (deleteClicked)="deleteDataFeed()"
  objectAction="Delete">
</app-delete-object-modal>
```

File: components/automate-ui/src/app/entities/destinations/destination.model.ts (hunk 6 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/entities/destinations/destination.model.ts:14]: Done

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html (hunk 7 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html:79]: Done

**Comment by shaik80** [comment]
**In webpage**
Screenshot 2021-09-07 at 4 10 50 PM
**Figma design** 
Screenshot 2021-09-07 at 4 15 06 PM

make sure to match same as Figma design

**Comment by shivangi-singh1** [comment]
> **In webpage**
> Screenshot 2021-09-07 at 4 10 50 PM
> **Figma design**
> Screenshot 2021-09-07 at 4 15 06 PM
> 
> make sure to match same as Figma design

Done

**Comment by shivangi-singh1** [comment]
> Please update the PR description accordingly

Done

**Review by vivek-yadav — APPROVED** [review]

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html [hunk 8]
```diff
@@ -0,0 +1,79 @@
+<div class="page-body">
+  <div id="servers-table-container" *ngIf="destinations.length != 0">
+    <chef-table class="node-list">
+      <caption class="visually-hidden">Node List Table with Node States</caption>
+      <chef-thead>
+          <chef-tr>
+              <!-- <chef-th  scope="col" class="delete">
+                  <chef-checkbox class="delete-checkbox row"> </chef-checkbox>
+                </chef-th> -->
```
Comment (shaik80) [inline components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html:9]: remove comments

**Review by shaik80 — APPROVED** [review]

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html (hunk 8 above)
Comment (shivangi-singh1) [reply components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html:9]: Done

**Review by MadhviAtroliya — APPROVED** [review]

File: e2e/cypress/integration/ui/settings/data-feed/data-feed-table.spec.ts [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        cy.get('app-notification.info chef-icon').click();
+    });
+
+
+      it('check if clicking on Cancel works', () => {
+        cy.get('chef-tr').contains(delete_name).parent().parent()
+        .find('.mat-select-trigger').click();
+        cy.get('[data-cy=remove-data-feed]').click();
+        cy.get('app-data-feed-table chef-modal').find('chef-button').contains('Cancel').click();
+    });
```
Comment (vivekshankar1) [inline e2e/cypress/integration/ui/settings/data-feed/data-feed-table.spec.ts:86]: check indentation here and other places in this file

File: components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html [hunk 10]
```diff
@@ -0,0 +1,75 @@
+<div class="page-body">
+  <div id="servers-table-container" *ngIf="destinations.length != 0">
+    <chef-table class="node-list">
+      <caption class="visually-hidden">Node List Table with Node States</caption>
+      <chef-thead>
+          <chef-tr>
+
+                <chef-th scope="col" class="node-name sort"  (click)="onToggleSort('name')">
```
Comment (vivekshankar1) [inline components/automate-ui/src/app/page-components/data-feed-table/data-feed-table.component.html:8]: please check indentation in this file

**Review by vivekshankar1 — APPROVED** [review]
Changes looks good

**Pull request merged**